controllers/tab/publicationEntry/form/CitationsForm.inc.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+/**
+ * @file controllers/tab/publicationEntry/form/CitationsForm.inc.php
+ *
+ * Copyright (c) 2014-2017 Simon Fraser University
+ * Copyright (c) 2003-2017 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class CitationsForm
+ * @ingroup controllers_tab_publicationEntry_form
+ *
+ * @brief Displays a citations form.
+ */
+
+import('lib.pkp.classes.form.Form');
+
+class CitationsForm extends Form {
+
+	/** @var int The context id */
+	var $_contextId;
+
+	/** @var Submission */
+	var $_submission;
+
+	/** @var int The current stage id, WORKFLOW_STAGE_ID_ */
+	var $_stageId;
+
+	/** @var int The current tab position */
+	var $_tabPos;
+
+	/** @var array Parameters to configure the form template. */
+	var $_formParams;
+
+	/**
+	 * Constructor.
+	 * @param $submission object
+	 * @param $stageId integer
+	 * @param $tabPos integer
+	 * @param $formParams array
+	 */
+	function __construct($submission, $stageId, $tabPos, $formParams = null) {
+		parent::__construct('controllers/tab/publicationEntry/form/citationsForm.tpl');
+
+		$this->_submission = $submission;
+		$this->_stageId = $stageId;
+		$this->_tabPos = $tabPos;
+		$this->_formParams = $formParams;
+
+		$request = Application::getRequest();
+		$context = $request->getContext();
+		$this->_contextId = $context->getId();
```

Comment:
Should we instead get the context from the submission? I'm thinking of a possible use-case where a submission entry modal might be launched from a submissions list in the admin area with no context...

Reply:
Yes, sounds good! :-)